homeassistant/components/enocean/__init__.py

```diff
@@ -123,10 +142,15 @@ def __init__(self):
         ENOCEAN_DONGLE.register_device(self)
```

Comment:
We shouldn't store the entity in a shared container. We should use our dispatch helper to signal the correct entity from the component.

Please refactor this either here or in a new PR that we merge first.